### ISSUE TITLE
Lantern doesn't run on XP (sometimes)

### ISSUE THREAD

**Issue by uaalto** [getlantern/lantern#3077]
Seems caused by **panicwrap** package. Getting rid of it seems to fix it. There are two options here:

* Remove panicwrap usage under Windows XP (not very satisfactory given the amount of users that run this OS).
* Try to fix panicwrap itself (may be due to an OS limitation or a Go bug, or very time-consuming to catch).

**Comment by fffw** [comment]
I built two versions of 2.0.0 with and without `-Hwindowsgui` respectively, and run them on an XP SP3 vm. The one without console always starts up and works well, will another one never starts up. I didn't see latter in task manager.

Both binaries can run on Windows 7 without issue. Binary without `panicwrap` works well on XP.

It's not easy to spot the error returned by `panicwrap` though.

**Comment by uaalto** [comment]
I have made some progress on this issue. Apparently is two different issues at once. Panicwrap is not the cause. I'll open another issue for the bug I've identified, and try to verify the second part.

**Comment by uaalto** [comment]
I forgot to tell that the bug can be triggered with and without panicwrap and/or windowsgui flag.

**Comment by uaalto** [comment]
The first issue, referenced in #3086 was:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x53633b]

goroutine 484 [running]:
sync/atomic.LoadUint64(0x117acec4, 0x53aa1a, 0xc)
        /go/src/sync/atomic/asm_386.s:165 +0xb
github.com/getlantern/detour.(*Conn).anyDataReceived(0x117aceb0, 0x117c8f40)
        /lantern/src/github.com/getlantern/detour/detour.go:199 +0x32
github.com/getlantern/detour.func┬╖008()
        /lantern/src/github.com/getlantern/detour/detour.go:145 +0xf2
created by github.com/getlantern/detour.func┬╖010
        /lantern/src/github.com/getlantern/detour/detour.go:150 +0x63f

goroutine 1 [syscall, locked to thread]:
syscall.(*Proc).Call(0x1162d980, 0x115dff74, 0x2, 0x2, 0x7565a0, 0x8d3134, 0x0, 0x0)
        /go/src/syscall/dll_windows.go:132 +0x27b
syscall.(*LazyProc).Call(0x11557a60, 0x115dff74, 0x2, 0x2, 0x0, 0x0, 0x0, 0x0)
        /go/src/syscall/dll_windows.go:279 +0x5f
github.com/getlantern/systray.nativeLoop()
        /lantern/src/github.com/getlantern/systray/systray_windows.go:49 +0xba
github.com/getlantern/systray.Run(0x8d31ec)
        /lantern/src/github.com/getlantern/systray/systray.go:56 +0x82
main.runOnSystrayReady(0x8d31ec)
        /lantern/src/github.com/getlantern/flashlight/systray.go:14 +0x28
main.main()
        /lantern/src/github.com/getlantern/flashlight/flashlight.go:129 +0x4c
```

The second is:
```
Sep 02 18:41:29.169 - 0m35s ERROR flashlight.client: server.go:1777 http: panic serving 127.0.0.1:4239: runtime error: invalid memory address or nil pointer dereference
goroutine 3052 [running]:
net/http.func·011()
        /go/src/net/http/server.go:1130 +0xa8
io.Copy(0x31595f10, 0x12173368, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        /go/src/io/io.go:362 +0x17f
net.genericReadFrom(0x31581198, 0x11f5aa60, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        /go/src/net/net.go:404 +0x92
net.(*TCPConn).ReadFrom(0x11f5aa60, 0x0, 0x0, 0x315856c8, 0x11f5aa60, 0x0, 0x0)
        /go/src/net/tcpsock_posix.go:72 +0xbc
io.Copy(0x31581198, 0x11f5aa60, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        /go/src/io/io.go:358 +0xf4
github.com/getlantern/flashlight/client.pipeData(0x31580ac0, 0x11f5aa60, 0x0, 0x0, 0x11d7dce0)
        /lantern/src/github.com/getlantern/flashlight/client/handler.go:136 +0x155
github.com/getlantern/flashlight/client.(*Client).intercept(0x1155c780, 0x315855a8, 0x1158b730, 0x11a34a80)
        /lantern/src/github.com/getlantern/flashlight/client/handler.go:105 +0x596
github.com/getlantern/flashlight/client.(*Client).ServeHTTP(0x1155c780, 0x315855a8, 0x1158b730, 0x11a34a80)
        /lantern/src/github.com/getlantern/flashlight/client/handler.go:30 +0x14e
net/http.serverHandler.ServeHTTP(0x116a9980, 0x315855a8, 0x1158b730, 0x11a34a80)
        /go/src/net/http/server.go:1703 +0x145
net/http.(*conn).serve(0x11859440)
        /go/src/net/http/server.go:1204 +0x9d8
created by net/http.(*Server).Serve
        /go/src/net/http/server.go:1751 +0x2ce
```

**Comment by uaalto** [comment]
This looks very much like a memory alignment problem as well, but buried some where I'm trying to find.

**Comment by uaalto** [comment]
Ok! Once figured out the first issue, it appears that the real cause is __panicwrap__

**Comment by uaalto** [comment]
Is this in production already?

**Comment by myleshorton** [comment]
Yes I believe so @uaalto, although now I'm not seeing the merge to release-2.0.1 so not completely sure.

**Comment by fffw** [comment]
It's merged in https://github.com/getlantern/lantern/pull/3097 right?

**Comment by uaalto** [comment]
Yes. I'll keep it here anyway until it's in valencia, just in case.

**Comment by myleshorton** [comment]
> It's merged in https://github.com/getlantern/lantern/pull/3097 right?

Looks like it yeah! I think it was also already merged to valencia in #3087

**Comment by uaalto** [comment]
They are different, but they are both in valencia: 
https://github.com/getlantern/lantern/commit/46c4fa73cd3f674103d777c55370e969fda5d173
https://github.com/getlantern/lantern/commit/f3e4c3536fab6bbfea2a475519f6710f1b33ddb3
Closing! :)
